tests/ftqc/test_decomposition.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Unit tests for the decomposition transforms in the FTQC module"""
+
+from functools import partial
+
+import networkx as nx
+import numpy as np
+import pytest
+
+import pennylane as qml
+from pennylane.devices.qubit.apply_operation import apply_operation
+from pennylane.ftqc import (
+    GraphStatePrep,
+    RotXZX,
+    cond_measure,
+    convert_to_mbqc_gateset,
+    diagonalize_mcms,
+    measure_arbitrary_basis,
+    measure_x,
+)
+from pennylane.ftqc.decomposition import _rot_to_xzx, mbqc_gate_set
+from pennylane.ops.op_math.decompositions.single_qubit_unitary import _get_xzx_angles
+
+
+class TestGateSetDecomposition:
+    """Test decomposition to the MBQC gate-set"""
+
+    def test_rot_to_xzx_decomp_rule(self):
+        """Test the _rot_to_xzx rule"""
+        phi, theta, omega = 1.39, -0.123, np.pi / 7
+
+        mat = qml.Rot.compute_matrix(phi, theta, omega)
+        a1, a2, a3, _ = _get_xzx_angles(mat)
+
+        with qml.queuing.AnnotatedQueue() as q:
+            _rot_to_xzx(phi, theta, omega, wires=0)
+
+        ops = qml.tape.QuantumScript.from_queue(q).operations
+        assert len(ops) == 1
+        assert isinstance(ops[0], RotXZX)
+        assert np.allclose(ops[0].data, (a3, a2, a1))
```

Comment:
Maybe my brain is tired, but why do we expect the order here to be 321, not 123?

Reply:
It's just a weird quirk of the order things were done in `_get_xzx_angles`. I didn't modify any of the code there, just split it up, and it calculates `phi, theta, lam` and then uses `lam, theta, phi`. But I don't see any reason we can't update it, and it looks like @astralcai is reversing the order anyway, so we'll just change this.